File: src/main/java/lazy/helper/handlers/FuelHandler.java
package lazy.helper.handlers;

import java.util.HashMap;
import java.util.Map;

import lazy.helper.objects.ItemBase;
import net.minecraft.item.ItemStack;
import cpw.mods.fml.common.IFuelHandler;
import cpw.mods.fml.common.registry.GameRegistry;

public class FuelHandler implements IFuelHandler {
	
	public FuelHandler () {
		GameRegistry.registerFuelHandler(this);
	}
	
	Map<ItemStack, Integer> fuels = new HashMap<ItemStack, Integer>();
	
	public void addFuel (ItemStack fuel, int time) {
		fuels.put(fuel, time);
	}

	@Override
	public int getBurnTime(ItemStack fuelStack) {
		Integer time = fuels.get(fuelStack);
		return (time != null) ? time : 0;
	}

}

File: src/main/scala/geek/lawsof/physics/lib/handler/FuelHandler.scala
package geek.lawsof.physics.lib.handler

import cpw.mods.fml.common.IFuelHandler
import net.minecraft.item.ItemStack
import cpw.mods.fml.common.registry.GameRegistry

/**
 * Created by anshuman on 28-05-2014.
 */
class FuelHandler extends IFuelHandler {
  GameRegistry.registerFuelHandler(this)

  var fuels: Map[ItemStack, Int] = Map.empty

  def add(stack: ItemStack, items: Int) = fuels += (stack -> items * 200)

  override def getBurnTime(fuel: ItemStack): Int = fuels(fuel)
}
